### PR TITLE
Add event notification on cluster

### PR DESCRIPTION
**Pull request by ermel272**
### Description
Moved notification pushing logic to its own class and added a notification that occurs on clustering.
Notification for the event clustering uses the same ID to update the same notification.

### Changelog
- Factor out notification logic from TweetFeedActivity
- Add event notification to WeightedNodeRenderer
- misc reordering of public/protected/private methods in certain edited classes

### Addresses Issues
#100

### CODE REVIEW

**Comment by bazadactyl** [comment]
Good job with the notification handler class +1

File: app/src/main/java/com/dankideacentral/dic/activities/TweetFeedActivity.java [hunk 1]
```diff
@@ -172,100 +164,29 @@ public boolean handleMessage(Message msg) {
                 String tweetUserName = tweet.getUser().getName();
                 long tweetUserId = tweet.getUser().getId();
                 if(friends.contains(tweetUserId)) {
-                    sendNotification("Your friend " + tweetUserName + " tweeted near you!");
+                    notificationHandler.sendNotification("Your friend " + tweetUserName + " tweeted near you!",
+                            notificationId++);
```
Comment (bazadactyl) [inline app/src/main/java/com/dankideacentral/dic/activities/TweetFeedActivity.java:168]: Do you think the TweetFeedActivity should be concerned with notification ID's any more? Or should the NotificationHandler deal with them?

**Review by srowhani — APPROVED** [review]
Looks good.

File: app/src/main/java/com/dankideacentral/dic/activities/TweetFeedActivity.java (hunk 1 above)
Comment (ermel272) [reply app/src/main/java/com/dankideacentral/dic/activities/TweetFeedActivity.java:168]: I figured it should be the responsibility of the calling class to handle managing identification numbers, similar to how the notification interface 'notificationManager.notify()' takes an ID as input (ie it doesn't manage it for you).

**Pull request merged**